api/src/test/java/com/messagebird/MessageBirdClientTest.java

```diff
@@ -565,6 +565,70 @@ public void testCreateTranscription() throws UnauthorizedException, GeneralExcep
         assertEquals(response.getData().get(0).getCreatedAt(), transcriptionResponse.getData().get(0).getCreatedAt());
     }
 
+    @Test
+    public void testListRecordings() throws UnauthorizedException, GeneralException {
+        final RecordingResponse recordings = TestUtil.createRecordingResponseList();
+
+        MessageBirdService messageBirdServiceMock = mock(MessageBirdService.class);
+        MessageBirdClient messageBirdClientInjectMock = new MessageBirdClient(messageBirdServiceMock);
+
+        String url = String.format(
+                "%s%s/%s%s/%s%s",
+                VOICE_CALLS_BASE_URL,
+                VOICECALLSPATH,
+                "ANY_CALL_ID",
+                LEGSPATH,
+                "ANY_LEG_ID",
+                RECORDINGPATH
+        );
+        when(messageBirdServiceMock.requestList(url, 0, 0, RecordingResponse.class))
+                .thenReturn(recordings);
+
+        final RecordingResponse response = messageBirdClientInjectMock
+                .listRecordings("ANY_CALL_ID", "ANY_LEG_ID", 0, 0);
+        verify(messageBirdServiceMock, times(1)).requestList(url, 0, 0, RecordingResponse.class);
+        assertNotNull(response);
+        for(int i = 0; i < response.getData().size() ; i++) {
+            assertEquals(response.getData().get(i).getId(), recordings.getData().get(i).getId());
```

Comment:
instead of these assertEquals you can use assertReflectionEquals(response.getData().get(i), recordings.getData().get(i)); 
We added specific library(unitils-core) for test scope. it asserts recursively.